### ISSUE TITLE
New version of this package

### ISSUE THREAD

**Issue by ewengillies** [bitly/data_hacks#45]
Hello devs,

This package has been a part of my workflow for several years now, mainly since I spend most of my time on the command line.  I see its not really maintained anymore.  I would like to take responsibility for it if no one minds, mostly so I can get it working with python 3 (and have this version on PyPI) and add some features.

If this sounds okay, I propose one of three ways to transition this (in order of my preference):

1. I fork the package and keep the same name.  In this scenario, I'd like access to the `data_hacks` package on PyPI so I can upload the python 3 version and keep it up-to-date.
2. I fork the package and use a new name (e.g. `data_hacks_3` or `data_cli`).  This would be more like making my own package with this package as a starting point.  In this scenario, I would just need your permission, I'll handle the rest.
3. I offer my support to the main fork of the package.  I think this solution would cause the most overhead for you all, so that's why I've listed this solution last.

Let me know which sounds best for you. Thanks!
Ewen

**Comment by mccarthyryanc** [comment]
@ewengillies , I've been waiting YEARS for updates to this repo. Given the level of responsiveness, I'd say you're better off going with option 2. No LICENSE file on the repo, but there is a license within the code comments. I'm not a lawyer, but If you carry that same license in a new repo (and refer back to this repo), I think you would be okay.

**Comment by fschaefer** [comment]
@ewengillies I agree with @mccarthyryanc : the code is Apache licensed. Just find a proper name and go ahead with option 2 and fork the repository.

**Comment by ewengillies** [comment]
@fschaefer and @mccarthyryanc, thanks for the advice.  I had a bit more time a month ago, but I know I'll find time for this.  I don't think it will be that much of an effort to do a python 3 port, hopefully I'll find time to revamp the code a bit and add some features.  I'll update on this issue when I have a stable release.

**Comment by ewengillies** [comment]
just as a quick poll, what features do you use from this package?

**Comment by mccarthyryanc** [comment]
@ewengillies , `bar_chart` and `histogram` are the main ones I use. Updating for Python3 was pretty simple on these utilities too. When you fork, I'd be happy to add some PRs with updates.

**Comment by tw-xren** [comment]
Hi @ewengillies , Have you get a chance to work on option 2? I was thinking to make a python3 port for this tool recently and then find this issue. If not I can proceed if you don't have something that is already work in progress.

**Comment by ewengillies** [comment]
@xiaojun-tw

It was a weekend project that never took off.  Feel free to do the port, and keep us on this thread posted please!  Let me know if I can help at all.

**Comment by mccarthyryanc** [comment]
While not an exact replacement, I've been using [visidata](https://www.visidata.org/) in place of `data_hacks`. It does quite a lot and includes tools for [statistics and histograms](https://www.visidata.org/docs/group/).

**Comment by nicholasren** [comment]
Added python3 support and released at https://pypi.org/project/data-hacks3/
